### PR TITLE
Added tests for `goprocess`.

### PR DESCRIPTION
**Pull request by yarcat**
The tests verify #123 fix and prevent future regressions.

### CODE REVIEW

**Review by tklauser — COMMENTED** [review]
Why do we need a goroutine for each fake process? Wouldn't it be simpler to create "fake" `pss` and `want` slices and use them? To me it seems, for testing purposes the processes don't need to be backed by actual processes/goroutines which makes the test code a lot simpler and easier to read. It would also be nice to test a case where `pss` contains not only Go processes (i.e. for which `isGoFunc` returns `false`).

**Comment by yarcat** [comment]
@tklauser Thank you for pointing to those empty go-routines. The initial idea was to build a semaphore for `isGo` with them, but I never finished it as the test started to catch the bugs, and I never had a chance to get back to it.

I did a major refactoring of the test and still decided to implement a semaphore to ensure workers don't finish too fast. This could be an overkill, but I feel it safer this way.

**Review by tklauser — CHANGES_REQUESTED** [review]
Sorry, I still don't understand why we would need concurrency in the test itself. We're testing the concurrency in `FindAll`/`findAll` and IMO it would be way easier to understand the test if it just constructed some example `pss` and expected `[]P` slices (some larger than `concurrencyLimit` to test for #123) and checked them against `FindAll` sequentially. I don't think we need to test for concurrent calls to `FindAll`. Care to explain why you think the semaphore-synced `isGo` etc. are needed in the tests?

**Comment by yarcat** [comment]
There is no real concurrency in the tests. All tests but the last one wouldn't block on anything.

The last test is a bit special. It actually does two things:
1) It tests an issue #123 
2) It guarantees max concurrency is reached and all workers will be activated [ ... oups ... ] Actually, while writing this, I realized that it's Go that guarantees all of the workers will be activated.

You're right! Thanks for challenging me on this. I'll drop the semaphore.

**Comment by yarcat** [comment]
@tklauser Btw, why did you decide that it was testing `findAll` concurrency? All the semaphoring magic happened inside a single test only. And it ensured `findAll` internally enables all of its workers before proceeding. Only now during writing my previous comment I realized, that go does round-robin, and thus all of the `concurrencyLimit` workers will have a chance to execute `isGo`.
But it has nothing to do with testing `findAll` concurrency.

File: goprocess/gp_test.go [hunk 1]
```diff
@@ -1,9 +1,93 @@
...
+	"github.com/keybase/go-ps"
+)
 
 func BenchmarkFindAll(b *testing.B) {
 	for ii := 0; ii < b.N; ii++ {
 		_ = FindAll()
 	}
 }
+
+type fixtureFunc func() (pss []ps.Process, want []P, fn isGoFunc, done func())
```
Comment (tklauser) [inline goprocess/gp_test.go:17]: Nit: this is now unused an can be dropped as well.

Comment (yarcat) [reply goprocess/gp_test.go]: Done

File: goprocess/gp_test.go [hunk 2]
```diff
@@ -1,9 +1,91 @@
...
+		return
+	}
+}
+
+func fakeProcessesWithPIDs(pids ...int) (p []ps.Process) {
+	for _, pid := range pids {
+		p = append(p, fakeProcess{pid: pid})
+	}
+	return
+}
```
Comment (tklauser) [inline goprocess/gp_test.go:82]:
```suggestion
func fakeProcessesWithPIDs(pids ...int) []ps.Process {
	p := make([]ps.Process, 0, len(pids))
	for _, pid := range pids {
		p = append(p, fakeProcess{pid: pid})
	}
	return p
}
```

**Review by tklauser — APPROVED** [review]
Thanks

**Pull request merged**